src/main/java/com/photolocos/enterprise/dao/LocationDAOStub.java

```diff
@@ -42,20 +43,20 @@ public LocationDTO fetchUserLocation() {
     }
 
     @Override
-    public LocationDTO fetchByCoordinates(String longitude, String latitude) {
+    public LocationDTO fetchByCoordinates(double longitude, double latitude) {
         List<LocationDTO> locationsCollection = new ArrayList<>(locations.values());
         LocationDTO neededLocation = new LocationDTO();
 
         for (LocationDTO loc : locationsCollection) {
-            if (loc.getLatitude().equals(latitude) && loc.getLongitude().equals(longitude)) {
+            if (loc.getLatitude() == latitude && loc.getLongitude() == longitude) {
                 neededLocation = loc;
             }
         }
         return neededLocation;
     }
 
     @Override
-    public LocationDTO fetchByArea(String state, String city) {
+    public LocationDTO fetchByStateAndCity(String state, String city) {
         List<LocationDTO> locationsCollection = new ArrayList<>(locations.values());
         LocationDTO neededLocation = new LocationDTO();
 
```

Comment:
Based on the method call from the controller and the parameters, this should be fetchByStateAndCity